src/hooks/useApproveAndDeposit.ts

```diff
@@ -87,9 +87,12 @@ export function useApproveAndDeposit(
       )
         throw new Error("Swap contract is not loaded")
 
+      const poolBaseTokens = pool?.isMetaSwap
```

Comment:
Additional complexity accounts for token composition depending on the type of pool. @hammeiam